sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/NuGetMetadataReference.cs

```diff
@@ -109,110 +92,146 @@ private static MetadataReference[] CreateReferences(string packageId, string pac
 
         #endregion Helpers
 
-        private static Lazy<Dictionary<string, MetadataReference[]>> lazyFluentAssertions =
-            CreateLazy("FluentAssertions", "4.19.4", Constants.NuGetLatestVersion);
+        #region Package installation and update helpers
 
-        private static Lazy<Dictionary<string, MetadataReference[]>> lazyMicrosoftAspNetCoreMvcCore =
-            CreateLazy("Microsoft.AspNetCore.Mvc.Core", "2.0.4", Constants.NuGetLatestVersion);
+        private static void EnsurePackageIsInstalled(string packageId, string packageVersion)
+        {
+            if (packageVersion == Constants.NuGetLatestVersion)
+            {
+                if (IsCheckForLatestPackageRequired(packageId))
```

Comment:
IMO you are still missing the case where we have deleted the `packages` folder. I think that in addition to the date check you should also ensure that you find at least one folder for the given package id.

Reply:
Ah wait, if I understood correctly latter comments this is handled.

Reply:
It is. I've manually test by deleting the entire _packages_ folder, by deleting specific packages, and by manually updating/deleting the timestamp files.